internals/secrethub/audit.go

```diff
@@ -82,30 +130,161 @@ func (cmd *AuditCommand) run() error {
 			return err
 		}
 
-		fmt.Fprint(tabWriter, strings.Join(row, "\t")+"\n")
-
-		if i == cmd.perPage {
-			err = tabWriter.Flush()
-			if err != nil {
-				return err
-			}
-			i = 0
+		formattedRow, err := formatter.formatRow(row)
+		if err != nil {
+			return err
+		}
 
-			// wait for <ENTER> to continue.
-			_, err := ui.Ask(cmd.io, "Press <ENTER> to show more results. Press <CTRL+C> to exit.")
-			if err != nil {
-				return err
-			}
-			fmt.Fprint(tabWriter, header)
+		if paginatedWriter.IsClosed() {
+			break
+		}
+		_, err = fmt.Fprintln(paginatedWriter, formattedRow)
+		if err != nil {
+			return err
 		}
 	}
+	return nil
+}
+
+type tableFormatter interface {
```

Comment:
The idea behind 'passing strings around' was that this way the steps the data goes through are more clear `formatting -> writing`. But indeed making the `formatters` directly write to a `writer` would simplify and decouple the abstraction.

Reply:
I don't think it is tightly coupled, it's just named poorly :sweat_smile:. It is used for formatting the output to both the `table` and the `json` format. I named it `tableFormatter` since it takes the result of the `auditTable` type. I built it this way to make it more reusable:
The `auditTable` returns a table row as a `[]string`, the `tableFormatter` formats it to either `json` or `table` format depending on which instance of the interface it is. This way the `tableFormatter` can be used to format all kinds of table rows not just audit entries.

Reply:
I'm not sure what to name it though.

Reply:
Haha yes, it's probably a naming issue. Consider this: `table` is the format, just like `json` is the format. What we're actually doing is processing a `list` of `audit events`. So something like `listFmt`, `listFormatter`, `eventListFormatter` or `auditEventListFormatter` could work?

Reply:
Thanks for the suggestions!